tests/test_main.cpp

```diff
@@ -32,41 +67,55 @@ void init_global_game_state() {
     get_options().load();
     init_colors();
 
+    const std::vector<std::string> mods = extract_mod_selection( arg_vec );
+
     g = new game;
 
     g->load_static_data();
-    g->load_core_data();
-    DynamicDataLoader::get_instance().finalize_loaded_data();
 
     world_generator->set_active_world(NULL);
     world_generator->get_all_worlds();
-    WORLDPTR test_world = world_generator->make_new_world( false );
+    WORLDPTR test_world = world_generator->make_new_world( mods );
     assert( test_world != NULL );
     world_generator->set_active_world(test_world);
     assert( world_generator->active_world != NULL );
 
+    g->load_world_modfiles( world_generator->active_world, true );
+
     g->u = player();
     g->u.create(PLTYPE_NOW);
+
     g->m = map( static_cast<bool>( ACTIVE_WORLD_OPTIONS["ZLEVELS"] ) );
 
     g->m.load( g->get_levx(), g->get_levy(), g->get_levz(), false );
 }
 
+std::vector<const char *> arg_vector( int argc, const char *argv[] )
```

Comment:
You can use the vector constructor that takes two iterators (pointers like `argv` can be used as iterators) instead of this function:

```C++
std::vector<const char *> arg_vec(argv, argv + argc);
```